packages/dht/src/connection/WebRTC/WebRtcConnector.ts

```diff
@@ -0,0 +1,274 @@
+            })
+        } else {
+            connection.once(IWebRtcEvent.LOCAL_DESCRIPTION, async (description, _type) => {
+                await remoteConnector.sendRtcAnswer(this.ownPeerDescriptor!, description, connection.connectionId.toString())
+            })
+        }
+        connection.on(IWebRtcEvent.LOCAL_CANDIDATE, async (candidate, mid) => {
+            await remoteConnector.sendIceCandidate(this.ownPeerDescriptor!, candidate, mid, connection.connectionId.toString())
+        })
+        connection.on(ConnectionEvents.CONNECTED, () => {
+            this.emit(ConnectionSourceEvents.CONNECTED, connection)
+        })
+        connection.start(offering)
+        if (offering === false && sendRequest) {
+            remoteConnector.requestConnection(this.ownPeerDescriptor!, connection.connectionId.toString())
+                .catch(() => { })
+        }
+    }
+
+    public isOffering(myId: string, theirId: string): boolean {
+        return this.offeringHash(myId + theirId) < this.offeringHash(theirId + myId)
+    }
+
+    private offeringHash(idPair: string): number {
+        const buffer = crypto.createHash('md5').update(idPair).digest()
+        return buffer.readInt32LE(0)
+    }
+
+    bindListeners(incomingMessageHandler: TODO, protocolVersion: string): void {
+        // set up normal listeners that send a handshake for new connections from webSocketConnector
+        this.on(ConnectionSourceEvents.CONNECTED, (connection: IConnection) => {
+            connection.on(ConnectionEvents.DATA, async (data: Uint8Array) => {
+                const message = Message.fromBinary(data)
+                if (this.ownPeerDescriptor) {
+                    incomingMessageHandler(connection, message)
+                }
+            })
+            if (this.ownPeerDescriptor) {
+                logger.trace(`Initiating handshake with ${connection.getPeerDescriptor()?.peerId.toString()}`)
+                const outgoingHandshake: HandshakeMessage = {
+                    sourceId: this.ownPeerDescriptor.peerId,
+                    protocolVersion: protocolVersion,
+                    peerDescriptor: this.ownPeerDescriptor
+                }
+
+                const msg: Message = {
+                    appId: this.WEBRTC_CONNECTOR_APP_ID,
+                    messageType: MessageType.HANDSHAKE, messageId: 'xyz',
+                    body: HandshakeMessage.toBinary(outgoingHandshake)
+                }
+
+                connection.send(Message.toBinary(msg))
+                connection.sendBufferedMessages()
+            }
+        })
+    }
+
+    // IWebRTCConnector implementation
+
+    async requestConnection(request: WebRtcConnectionRequest, _context: ServerCallContext): Promise<Empty> {
+        setImmediate(() => this.onConnectionRequest(request.requester!))
+        const res: Empty = {}
+        return res
```

Comment:
```suggestion
        return {}
```

Same for lines 260, 266 and 272